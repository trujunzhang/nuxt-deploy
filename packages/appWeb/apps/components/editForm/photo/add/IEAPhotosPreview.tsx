import * as React from 'react'
import * as Telescope from '@appComponents/index'
import * as Types from '@app/types'

interface IIEAPhotosPreviewProps {
  previewDisabled: boolean
  previewImageSrc: string
  onUploadImagePressed: any
  onDeletePreviewPressed: () => any
}

export class IEAPhotosPreview extends React.Component<IIEAPhotosPreviewProps, {}> {
  renderDeleteForPreview() {
    return (
      <div className="photo-box_actions photo-box_actions--right">
        <a
          onClick={this.props.onDeletePreviewPressed}
          className="photo-box_action-link show-tooltip">
          <span
            id="icon_24X24"
            className="icon icon--24-trash icon--size-24 icon--inverse icon--fallback-inverted delete-photo">
            <svg className="icon_svg">
              <path d="M5 7V5a1 1 0 0 1 1-1h4V3a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v1h4a1 1 0 0 1 1 1v2H5zm13 12a3 3 0 0 1-3 3H9a3 3 0 0 1-3-3V8h12v11zm-8-8H9v8h1v-8zm5 0h-1v8h1v-8z" />
            </svg>
          </span>
          <span className="tooltip-wrapper">
            <span className="tooltip">{'Delete'}</span>
          </span>
        </a>
      </div>
    )
  }

  renderPreview() {
    const { previewImageSrc } = this.props
    // console.log('previewImageSrc: ', previewImageSrc)
    return (
      <ul className="uploaded-photos js-uploaded-photos photo-box-grid--inline-block photo-box-grid--wide">
        <li className="uploaded-photo">
          <div className="photo-caption-box photo-box js-photo-box photo-box--interactive">
            <img
              className="photo-caption-img js-photo-caption-img photo-box-img"
              src={previewImageSrc}
            />

            {this.renderDeleteForPreview()}
          </div>
        </li>
      </ul>
    )
  }

  render() {
    return (
      <div className="post-upload-container no-js-hidden">
        <Telescope.F8AppAlertSection />

        <div className="arrange arrange--middle finish-upload-header">
          <div className="arrange_unit">
            <div id="upload-title--success" className="upload-title biz-uploader">
              <span id="icon_18X18" className="icon icon--18-checkmark icon--size-18 icon--success">
                <svg className="icon_svg">
                  <path d="M7.232 14.273L1.93 8.97a1 1 0 1 1 1.413-1.414l3.89 3.89 7.424-7.426a1 1 0 0 1 1.414 1.414l-8.837 8.84z" />
                </svg>
              </span>
              <h3 className="alternate inline-block">
                {'Perfect!'}
                <b className="margin-left-4">{'Next, submit your photo.'}</b>
              </h3>
            </div>
          </div>

          {this.renderConfirmButton()}
        </div>

        {this.renderPreview()}
      </div>
    )
  }

  renderConfirmButton() {
    const { previewDisabled, onUploadImagePressed } = this.props
    return (
      <div className="arrange_unit arrange arrange--middle arrange--18 finish-social-media-container">
        <div className="arrange_unit">
          <button
            disabled={previewDisabled}
            onClick={onUploadImagePressed}
            className="ybtn ybtn-primary finish-upload-btn"
            type="submit"
            value="submit">
            <span>{'Finish'}</span>
          </button>
        </div>
      </div>
    )
  }
}
