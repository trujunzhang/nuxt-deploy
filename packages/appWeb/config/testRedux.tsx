import * as React from 'react'
import Telescope from '@appComponents/index'
import { I18nextProvider } from '@app/tools'

// get language from query parameter or url path
const lang = 'en'

import { getLocalI18n } from '@app/tools'
// import * as resources from '@assets/locales/i18nextLocalesLoader'

// export const mockedI18n = getLocalI18n(resources, (callback: any) => {
//   return 'en'
// })


// Store Initialization
// ------------------------------------
import createStore from '@src/redux/create'
const store = createStore()

import { createStore as RDCreateStore, Store } from 'redux';
import { Provider } from 'react-redux'
import withRedux, { MakeStore } from 'next-redux-wrapper';

import reducer from '@src/redux/rootReducer'
export type RootState = ReturnType<typeof reducer>;
/**
 * @param initialState The store's initial state (on the client side, the state of the server-side store is passed here)
 */
const makeStore: MakeStore = (initialState: RootState) => {
  return RDCreateStore(reducer, initialState);
};

export const testRedux = (key, showHeaderPanel = true) => {
  class Common extends React.Component {
    constructor(props) {
      super(props)
    }

  //   render() {
  //     const comp = Telescope.components[key]
  //     const children = React.createElement(comp, {})
  //     return (
  //       <Provider store={store}>
  //         <I18nextProvider i18n={mockedI18n}>
  //           <Telescope.components.Layout showHeaderPanel={showHeaderPanel}>
  //             {children}
  //           </Telescope.components.Layout>
  //         </I18nextProvider>
  //       </Provider>
  //     )
  //   }
  }

  return withRedux(makeStore)(Common)
}
