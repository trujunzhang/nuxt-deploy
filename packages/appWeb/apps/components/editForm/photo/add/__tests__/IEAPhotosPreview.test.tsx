import * as React from 'react'

// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer'

import { IEAPhotosPreview } from '../IEAPhotosPreview'
import { testRedux } from '@config/testRedux'

describe('Button', () => {
  it('render correctly for IEAPhotosPreview', () => {
    const children = (
      <IEAPhotosPreview
        previewDisabled={false}
        previewImageSrc={'blob:http://localhost:4000/ec21e2bb-3d16-4e48-a2e2-5f47329a7d71'}
        onUploadImagePressed={() => {}}
        onDeletePreviewPressed={() => {}}
      />
    )
    const renderPage = testRedux(children)
    const instance = renderer.create(renderPage)

    const page = instance.toJSON()
    expect(page).toMatchSnapshot()
  })
})
