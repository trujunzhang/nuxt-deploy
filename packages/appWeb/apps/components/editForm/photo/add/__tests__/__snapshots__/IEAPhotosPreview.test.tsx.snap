// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render correctly for IEAPhotosPreview 1`] = `
<div
  className="post-upload-container no-js-hidden"
>
  <div
    className="arrange arrange--middle finish-upload-header"
  >
    <div
      className="arrange_unit"
    >
      <div
        className="upload-title biz-uploader"
        id="upload-title--success"
      >
        <span
          className="icon icon--18-checkmark icon--size-18 icon--success"
          id="icon_18X18"
        >
          <svg
            className="icon_svg"
          >
            <path
              d="M7.232 14.273L1.93 8.97a1 1 0 1 1 1.413-1.414l3.89 3.89 7.424-7.426a1 1 0 0 1 1.414 1.414l-8.837 8.84z"
            />
          </svg>
        </span>
        <h3
          className="alternate inline-block"
        >
          Perfect!
          <b
            className="margin-left-4"
          >
            Next, submit your photo.
          </b>
        </h3>
      </div>
    </div>
    <div
      className="arrange_unit arrange arrange--middle arrange--18 finish-social-media-container"
    >
      <div
        className="arrange_unit"
      >
        <button
          className="ybtn ybtn-primary finish-upload-btn"
          disabled={false}
          onClick={[Function]}
          type="submit"
          value="submit"
        >
          <span>
            Finish
          </span>
        </button>
      </div>
    </div>
  </div>
  <ul
    className="uploaded-photos js-uploaded-photos photo-box-grid--inline-block photo-box-grid--wide"
  >
    <li
      className="uploaded-photo"
    >
      <div
        className="photo-caption-box photo-box js-photo-box photo-box--interactive"
      >
        <img
          className="photo-caption-img js-photo-caption-img photo-box-img"
          src="blob:http://localhost:4000/ec21e2bb-3d16-4e48-a2e2-5f47329a7d71"
        />
        <div
          className="photo-box_actions photo-box_actions--right"
        >
          <a
            className="photo-box_action-link show-tooltip"
            onClick={[Function]}
          >
            <span
              className="icon icon--24-trash icon--size-24 icon--inverse icon--fallback-inverted delete-photo"
              id="icon_24X24"
            >
              <svg
                className="icon_svg"
              >
                <path
                  d="M5 7V5a1 1 0 0 1 1-1h4V3a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v1h4a1 1 0 0 1 1 1v2H5zm13 12a3 3 0 0 1-3 3H9a3 3 0 0 1-3-3V8h12v11zm-8-8H9v8h1v-8zm5 0h-1v8h1v-8z"
                />
              </svg>
            </span>
            <span
              className="tooltip-wrapper"
            >
              <span
                className="tooltip"
              >
                Delete
              </span>
            </span>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
